pandas/tests/frame/test_stack_unstack.py

```diff
@@ -2018,3 +2018,18 @@ def test_stack_nan_level(self):
             ),
         )
         tm.assert_frame_equal(result, expected)
+
+    def test_unstack_categorical(self):
+        # GH 14018
+        idx = MultiIndex.from_product([["A"], [0, 1]])
+        df = DataFrame({"cat": pd.Categorical(["a", "b"])}, index=idx)
+        result = df.unstack()
+        expected = DataFrame(
+            [
```

Comment:
can you pass this as a dict or use from_arrays?  IIRC the constructor is wonky specifically if the first listlike is a categorical